llvm/lib/Analysis/ModuleSummaryAnalysis.cpp

```diff
@@ -175,6 +175,9 @@ static CalleeInfo::HotnessType getHotness(uint64_t ProfileCount,
 }
 
 static bool isNonRenamableLocal(const GlobalValue &GV) {
+  if (isa<Function>(&GV)) {
```

Comment:
Don't you care about its local linkage even it is a function?

Reply:
This is a proposal, to revert to the behavior before a recent change that changed how sections were carried in the IR for C++ functions (specifically, they used to be carried via attribute and not until later marked in the GV). Would you be able to look at the RFC, and specifically, the original patch that introduced this concept of not renaming locals with sections? It was added during the original ThinLTO development for MachO, and I'm wondering if you can help fill in the gaps on when this is needed for MachO. From a separate discussion with @MaskRay it seemed like this restriction might not be needed for ELF.

Reply:
Let me ask around to see if I can collect more context for macho (I don't believe this is macho specific but more low level firmware and expectations from legacy code). Let's continue discussion on the discourse RFC.

That said, if we go down the route of the patch proposed here, we need to rename the function by remove that `Local`.

Reply:
Sounds good on continuing discussion on RFC. Thanks in advance for digging into this.

> That said, if we go down the route of the patch proposed here, we need to rename the function by remove that Local.

I'm not sure why. It is still meant to return true if this is a non-renamable local symbol. We just want to fine tune the conditions.